datafusion/src/logical_plan/plan.rs

```diff
@@ -214,8 +220,14 @@ pub struct Extension {
     pub node: Arc<dyn UserDefinedLogicalNode + Send + Sync>,
 }
 
+impl PartialEq for Extension {
+    fn eq(&self, _other: &Self) -> bool {
+        todo!()
```

Comment:
I'm not sure if need to do something.

Reply:
> For the future, I think you can change
`PartialEq` isn't auto-trait,  so it can't be used as additional trait in a trait object